### PR TITLE
Updates for ACK Frequency Draft 2

### PR DESCRIPTION
**Pull request by nibanks**
Another change I wanted to get in before v2. Updates our ACK frequency usage to the latest draft version.

### CODE REVIEW

File: src/core/frame.c [hunk 1]
```diff
@@ -1227,13 +1245,15 @@ QuicAckFrequencyFrameDecode(
...
+    QUIC_ACK_FREQUENCY_EXTRAS Extras;
     if (!QuicVarIntDecode(BufferLength, Buffer, Offset, &Frame->SequenceNumber) ||
         !QuicVarIntDecode(BufferLength, Buffer, Offset, &Frame->PacketTolerance) ||
         !QuicVarIntDecode(BufferLength, Buffer, Offset, &Frame->UpdateMaxAckDelay) ||
-        !QuicUint8tDecode(BufferLength, Buffer, Offset, &Frame->IgnoreOrder) ||
-        Frame->IgnoreOrder > 1) { // IgnoreOrder should only be 0 or 1.
+        !QuicUint8tDecode(BufferLength, Buffer, Offset, &Extras.Value)) {
         return FALSE;
     }
+    Frame->IgnoreOrder = Extras.IgnoreOrder;
```
Comment (anrossi) [inline src/core/frame.c:1255]: Should the `FrameType` value be validated here to ensure it's the expected value?

**Review by anrossi — DISMISSED** [review]
Looks good, but should have a test too

File: src/core/frame.c (hunk 1 above)
Comment (anrossi) [reply src/core/frame.c:1255]: In `QUIC_ACK_FREQUENCY_EXTRAS`, `FrameType` is 6 bits, and I thought that was a useful value. If it's just a singular bit, I guess it doesn't need validation. Though, maybe the remaining 5 bits need to be validated as zero?

Comment (nibanks) [reply src/core/frame.c:1255]: Oops. `FrameType` should be `Reserved` and set to zero on write and ignored on read. So we have no functional issue here, but a variable name issue.

**Review by thhous-msft — APPROVED** [review]

**Review by anrossi — APPROVED** [review]

**Pull request merged**